.github/workflows/ci.yml

```diff
@@ -0,0 +1,134 @@
+name: ci
+
+on:
+  push:
+    branches:
+      - master
+      - /^\d+\.\d+\.\d(-alpha\.\d+|-beta\.\d+|-rc\.\d+)?$/
+    tags:
+      - "*"
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build-test:
+    name: Build and Test on node ${{ matrix.node_version }} and ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        node_version: ["10", "12"]
+        os: [ubuntu-latest, macOS-latest]
+    env:
+      LOGS_DIR: /tmp/ngx-form-errors/logs
+      LOGS_FILE: /tmp/ngx-form-errors/logs/build-perf.log
+      TZ: "Europe/Brussels"
+      MAIN_NODE_VERSION: "10"
+      MAIN_OS: "ubuntu-latest"
+    steps:
+      # See: https://github.com/marketplace/actions/checkout
+      - uses: actions/checkout@v2
+
+      # See: https://github.com/marketplace/actions/setup-node-js-for-use-with-actions
+      - name: Use Node.js ${{ matrix.node_version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node_version }}
+
+      - name: Create file & folder for GitHub Actions logs
+        run: |
+          # cfr scripts/_ghactions-group.sh
+          mkdir -p $LOGS_DIR
+          touch $LOGS_FILE
+
+      - name: Install npm 6.9.2
+        run: npm i -g npm@6.9.2
+
+      - name: List main variables
+        run: |
+          echo "Commit SHA  : ${GITHUB_SHA}"
+          echo "Reference   : ${GITHUB_REF}"
+          echo "Repository  : ${GITHUB_REPOSITORY}"
+          echo "Event       : ${GITHUB_EVENT_NAME}"
+          echo "Author      : ${GITHUB_ACTOR}"
+          NODE_VERSION="$(node -v)"
+          echo "Node version: $NODE_VERSION"
+          # This ensures that we are authenticated without requiring to have an actual .npmrc file within the project
+          echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}" >> ~/.npmrc
+
+      # See: https://help.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows#example-using-the-cache-action
+      - name: Cache node modules
+        uses: actions/cache@v1
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: ~/.npm # npm cache files are stored in `~/.npm` on Linux/macOS
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+
+      - name: Install dependencies
+        run: |
+          npm ci
+          npm run install:ci:demo:ng-previous
+          npm run install:ci:demo:ng-latest
+
+      # See: https://github.com/marketplace/actions/upload-artifact
+      - uses: actions/upload-artifact@v1
+        with:
+          name: ngx-form-errors-dist
+          path: dist
+        if: matrix.node_version == env.MAIN_NODE_VERSION && matrix.os == env.MAIN_OS && startsWith(github.ref, 'refs/tags/')
```

Comment:
Why do we also check here for `refs/tags`? Normally we should always do this job in every build right?

Reply:
Indeed, we could do it for every build 😊 
I don't think there is a limit in the number of artifacts we can store in GitHub Actions. I only know that ["Artifacts automatically expire after 90 days"](https://help.github.com/en/actions/configuring-and-managing-workflows/persisting-workflow-data-using-artifacts#downloading-or-deleting-artifacts)
I adapt this